pyuploadtool/releases_hosting_provider/github_releases.py

```diff
@@ -77,6 +78,19 @@ def create_release(self, metadata: ReleaseMetadata, artifacts):
 
         message = f"Build log: {metadata.build_log_url}"
 
+        if os.getenv("CHANGELOG_GENERATE", "").lower() == "true":
+
+            github_changelog = GitHubChangelogFactory(github_client=self.github_client, metadata=metadata)
+            metadata.changelog = github_changelog.get_changelog()
```

Comment:
What's the point of setting a changelog attribute at this point when you're not using it anyway?

The changelog should be generated in the GitHub actions build system. This side of the tool should then only render the given metadata. It should not add new information. The changelog might be used in other release hosting providers, too, but this would then require this one to run first.

Think: once the build system is done, metadata should not change any more.

Reply:
Ok, I have improved this part a bit more, by not editing the `metadata` in the release section. But however, I still haven't moved the changelog generation part to `build_system` because, 
* `GitHubChangelogFactory` requires a `GitHub` client instance. 
* The generation of changelog is optional, and will only happen if the `metadata.release_description` is not `None`.

I can move the `GitHubChangelogFactory` outside the `release_hosting_provider` but then, that will initialize the `GitHub` client again. Shall I proceed with that?